src/application/application.cpp

```diff
@@ -226,12 +226,23 @@ void Application::Predict() {
                       config_.precise_float_parser);
     TextReader<int> result_reader(config_.output_result.c_str(), false);
     result_reader.ReadAllLines();
-    std::vector<std::vector<int>> pred_leaf(result_reader.Lines().size());
+
+    auto nrow = result_reader.Lines().size();
+    auto ncol = 0;
+    if (nrow > 0) {
+      ncol = result_reader.Lines()[0].size();
```

Comment:
I'm unsure what this is doing. Don't we need to split by `\t` to know the number of columns?